Minutes 2022-03-02 (#288)
solid/authorization-panel: meetings/2022-03-02.md

# W3C Solid Community Group: Authorization Panel

> * Date: 2022-02-02T14:00:00Z

Author: ```suggestion
* Date: 2022-03-02T14:00:00Z
```

* No audio or video recording, or automated transcripts without consent. Meetings are transcribed and made public. If consent is withheld by anyone, recording/retention must not occur.
* Use panel chat and repository. Queue in call to talk.

### Participation and Code of Conduct
* [Join the W3C Solid Community Group](https://www.w3.org/community/solid/join), [W3C Account Request](http://www.w3.org/accounts/request), [W3C Community Contributor License Agreement](https://www.w3.org/community/about/agreements/cla/)
* [Solid Code of Conduct](https://github.com/solid/process/blob/master/code-of-conduct.md), [Positive Work Environment at W3C: Code of Ethics and Professional Conduct](https://www.w3.org/Consortium/cepc/)
* If this is your first time, welcome! please introduce yourself.

### Scribes
* Wouter
* Elf
* Matthieu

---

## Topics

* Default client based restrictions
* [Define responsibilities of Authorization Server #250](https://github.com/solid/data-interoperability-panel/issues/250)

### Default client based restrictions

URL: https://solid.github.io/authorization-panel/acp-specification/#client-matcher
URL: https://github.com/solid/authorization-panel/issues/80

> * Elf: ACP mentions the ACP Public Client which matches all clients. It's a way to explicitly say that any client is allowed. What is the difference between any Public Client and not mentioning it at all.

Reviewer: ```suggestion
* Elf: ACP mentions the ACP Public Client which matches all clients. It's a way to explicitly say that any client is allowed. What is the difference between any Public Client and not mentioning it at all?
```

> * Matthieu: Yes, there is not very much of a difference. If you did not mention any client than it would be the same. If you only match on the agent, lets say Bob, and there is not policy denying otherwise than bob has access mentioned.

Reviewer: ```suggestion
* Matthieu: Yes, there is not very much of a difference. If you did not mention any client then it would be the same. If you only match on the agent, lets say Bob, and there is no policy denying otherwise, then bob has access mentioned.
```

* Elf: I think I would like to propose to change that default, I think the default should be no client has access unless otherwise defined. I think that the least privileged access by default should be used. I see a problem with default access for every client. If you want, you can make it explicit.

> * Matthieu: Potentially client restrictions should be enforced not only by ACP but also via claims in the access token. For example https://github.com/solid/web-access-control-spec/blob/main/README-v0.5.0.md#adding-trusted-web-apps Here one specifies apps that you trust in your WebID, here Bob would only get access modes as Bob and intersection of access of app in trusted app listing.

Reviewer: ```suggestion
* Matthieu: Potentially client restrictions should be enforced not only by ACP but also via claims in the access token. For example, https://github.com/solid/web-access-control-spec/blob/main/README-v0.5.0.md#adding-trusted-web-apps. Here one specifies apps that you trust in your WebID, here Bob would only get access modes as Bob and intersection of access of app in trusted app listing.
```

> * Elf: We actually discussed it a little bit yesterday. I believe this trusted apps approach has a lot of problems including disclosing which apps are trusted by the user (public disclosure of trusted apps). Big privacy issue. In interop there is an alternative approach where the user delegates access to an app via a data grant. You could update an ACL via pushing a data grant to an AS.

Reviewer: ```suggestion
* Elf: We actually discussed it a little bit yesterday. I believe this trusted apps approach has a lot of problems including disclosing which apps are trusted by the user (public disclosure of trusted apps). Big privacy issue. In interop there is an alternative approach, where the user delegates access to an app via a data grant. You could update an ACL via pushing a data grant to an AS.
```

* ...: I believe the AS should be the one who sets ACLs on the RS. THe AS would have the knowledge of what the user trusts the app with. In this case the default is problematic: if there is no statement, every app owuld be allowed, which is no secure default.
* Matthieu: In a way, that means we need to make client IDs mandatory, because everything that does not support client restrictions would not work in an ecosystem where they are closed by default.
* Elf: The only exception case would be a kind of resource browsers.
* Matthieu: Are all implementations caught up with enforcing client_ids everywhere?
* Elf: We can keep in mind that there is a transition period, but aim towards no access by default.
* Matthieu: I agree on client restriction by default. 
* Justin: +1 client restriction by default

---

> * Matthieu: I do wonder if there is any consideration for authenticating without disclosing you issuer? Wondering if you could rely on some classic public/private key pair to do this.

Reviewer: ```suggestion
* Matthieu: I do wonder if there is any consideration for authenticating without disclosing your issuer? Wondering if you could rely on some classic public/private key pair to do this.
```

* Elf: Currently Solid-OIDC requires this to be public in the WebID.

> * Matthieu: It is a privacy issue if you want to use your WebID to use e.g. a government/social service, and you would want to hide this (assuming a government-tied issuer).

Reviewer: ```suggestion
* Matthieu: It is a privacy issue if you want to use your WebID to use, for instance, a government/social service, and you would want to hide this (assuming a government-tied issuer).
```

* Elf: I think this should be tracked as Solid-OIDC issue.

> * Justin: Given current events I think this is pretty on point use case. I think disclosing such associations might be an issue here.

Reviewer: ```suggestion
* Justin: Given current events, I think this is pretty on point use case. I think disclosing such associations might be an issue here.
```

> * Wouter: This is already central for Solid-OIDC. One of our clients pointed that disclosing the WebID/issuer relationship publicly as privacy risk.

Reviewer: ```suggestion
* Wouter: This is already central for Solid-OIDC. One of our clients pointed out that disclosing the WebID/issuer relationship publicly is a privacy risk.
```

* Matthieu: Wouldn't it be funny if we ended up going back to WebID/TLS?
* Elf: If we didn't rely on this public issuer in the Solid OIDC, and rather on a key, then you could pretty much have the keys published in the WebID, which would not disclose the issuer itself.

> * Justin: As I recall one of the big issues with WebID/TLS is key management (being seemless) and UX was horrendous in the browser. I was trying to find the quote, but Dimitri Zaguadulin showed me specs for in-browser support for digital wallets... I don't think it's for tomorrow, but that would really enable interesting solutions for in-browser/out-of-browser use of keys... I don't think we're there but we should keep an eye on it.

Reviewer: ```suggestion
* Justin: As I recall one of the big issues with WebID/TLS is key management (being seamless) and UX was horrendous in the browser. I was trying to find the quote, but Dimitri Zaguadulin showed me specs for in-browser support for digital wallets... I don't think it's for tomorrow, but that would really enable interesting solutions for in-browser/out-of-browser use of keys... I don't think we're there but we should keep an eye on it.
```

> * Elf: In the case where the resource server access policy requires a specific IdP then the resource server and authorization server have some pre-established trust with the IdP. We should take a use case where there is pre-established trust.

Reviewer: ```suggestion
* Elf: In the case where the resource server access policy requires a specific IdP, the resource server and authorization server have some pre-established trust with the IdP. We should take a use case where there is pre-established trust.
```

* Elf: I do not see why we would not add it to ACP. I think default should be taken into account when ACP policy is being resolved.
* Matthieu: Can we make this a default that applies to every authorization property?
* Elf: I think we can make an equivalent for each property, and then say that if a server does not support the default you have to use that property explicitly. This enables to make access explicitly open, even if not by default.

ACTION:
* Matthieu to create an issue on the authorization panel to make the `acp:client` property closed by default.
* Matthieu to create an issue on the authorization panel to make all authorization condition properties closed by default.
* Elf to create an issue about requirement to not disclose the issuer publicly in the Solid-OIDC panel and raise it in the panel on Monday the 7th.

### TS ACP in CSS

* Justin: I know you've been working on it, I want to ask about the progress.
* Matthieu: I released first iteration of the library. Didn't have time to work on it since then.

> * Justin: If CSS team prioritizes it, is you library ready to implement it?

Reviewer: ```suggestion
* Justin: If CSS team prioritizes it, is your library ready to implement it?
```

* ...: In short the UMA client can push some claims. Previously it was a DPoP bound access token, now it's an ID token. So the access token comes from the AS directly which is an UMA server and has pre-established trust with the resource server.
* Matthieu: Also, the UMA server can ask for some claims, right? So it's good for required credentials discovery?
* Elf: Yes, but we it would be involving some client interactions because the user should approve which credentials are pushed as a claim.
* Wouter: I think an important question about the UMA server is whether access should depend on specific resource access control. Right now it's exchange ID token for access token but then the RS decides what you can access on it.
* ...: It's important potentially to keep the two separated where you don't need to sync grants and ACLs.
* Elf: We could have a variant where based on the type of credentials would give you access.
* ...: For example, the client ID and the WebID are made available to the RS that receives the access token. Where the ACP resolver is executed is another question.

> * Matthieu: From what I know and from the ACP perspective, the ACLs that use the type of VC to access a resource does not include verification of that VC.

Reviewer: ```suggestion
* Matthieu: From what I know and from the ACP perspective, the ACLs that use the type of VC to access a resource do not include verification of that VC.
```
